src/openrct2/core/FileIndex.hpp

```diff
@@ -0,0 +1,286 @@
+#pragma region Copyright (c) 2017 OpenRCT2 Developers
+/*****************************************************************************
+ * OpenRCT2, an open source clone of Roller Coaster Tycoon 2.
+ *
+ * OpenRCT2 is the work of many authors, a full list can be found in contributors.md
+ * For more information, visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * A full copy of the GNU General Public License can be found in licence.txt
+ *****************************************************************************/
+#pragma endregion
+
+#pragma once
+
+#include <chrono>
+#include <string>
+#include <tuple>
+#include <vector>
+#include "../common.h"
+#include "File.h"
+#include "FileScanner.h"
+#include "FileStream.hpp"
+#include "Path.hpp"
+
+template<typename TItem>
+class FileIndex
+{
+private:
+    struct DirectoryStats
+    {
+        uint32 TotalFiles = 0;
+        uint64 TotalFileSize = 0;
+        uint32 FileDateModifiedChecksum = 0;
+        uint32 PathChecksum = 0;
+    };
+
+    struct ScanResult
+    {
+        DirectoryStats const Stats;
+        std::vector<std::string> const Files;
+
+        ScanResult(DirectoryStats stats, std::vector<std::string> files)
+            : Stats(stats),
+              Files(files)
+        {
+        }
+    };
+
+    struct FileIndexHeader
+    {
+        uint32          MagicNumber = 0;
```

Comment:
Can we embed header size into this struct? When doing various porting work, alignment rules may change and it would help invalidate the cache.

Reply:
OK - because I suppose it is likely people will unintentionally copy these local index files to other architectures.